noether/src/segmentation_server.cpp

```diff
@@ -0,0 +1,168 @@
+    nh_.param<int>("/mesh_segmenter_client_node/min_cluster_size", min_cluster_size, 500);
+    nh_.param<int>("/mesh_segmenter_client_node/max_cluster_size", max_cluster_size, 1000000);
+    nh_.param<double>("/mesh_segmenter_client_node/curvature_threshold", curvature_threshold, 0.3);
+    nh_.param<bool>("/mesh_segmenter_client_node/show_individually", show_individually, false);
+    nh_.param<bool>("/mesh_segmenter_client_node/save_outputs", save_outputs, false);
+
+    // Convert ROS msg -> PCL Mesh -> VTK Mesh
+    vtkSmartPointer<vtkPolyData> mesh;
+    pcl::PolygonMesh input_pcl_mesh;
+    pcl_conversions::toPCL(goal->input_mesh, input_pcl_mesh);
+    vtk_viewer::pclEncodeMeshAndNormals(input_pcl_mesh, mesh);//, 100, pcl::PointXYZ(0, 50.0, 50.0));
+    vtk_viewer::generateNormals(mesh);
+//    pcl::VTKUtils::convertToVTK(input_pcl_mesh, mesh);            // Converts w
+
+
+
+    // Step 2: Filter the mesh
+    // Create some pointers - not used since passing with VTK pipeline but useful for debugging
+    ROS_INFO("Beginning Filtering.");
+    vtkSmartPointer<vtkPolyData> mesh_in = mesh;
+    vtkSmartPointer<vtkPolyData> mesh_cleaned;
+    vtkSmartPointer<vtkPolyData> mesh_filtered1;
+    vtkSmartPointer<vtkPolyData> mesh_filtered2;
+
+    // Remove duplicate points (Note: this seems to cause problems sometimes)
+//    vtkSmartPointer<vtkCleanPolyData> cleanPolyData = vtkSmartPointer<vtkCleanPolyData>::New();
+//    cleanPolyData->SetInputData(mesh_in);
+//    cleanPolyData->Update();
+//    mesh_cleaned = cleanPolyData->GetOutput();
+
+    vtk_viewer::generateNormals(mesh_in);
+
+
+    if (goal->filter)
+    {
+      // Apply Windowed Sinc function interpolation Smoothing
+      vtkSmartPointer<vtkWindowedSincPolyDataFilter> smooth_filter1 =
+          vtkSmartPointer<vtkWindowedSincPolyDataFilter>::New();
+//      smooth_filter1->SetInputConnection(cleanPolyData->GetOutputPort());
+      smooth_filter1->SetInputData(mesh_in);
+      smooth_filter1->SetNumberOfIterations(20);
+      smooth_filter1->SetPassBand(0.1);
+      smooth_filter1->FeatureEdgeSmoothingOff();  // Smooth along sharp interior edges
+      smooth_filter1->SetFeatureAngle(45);        // Angle to identify sharp edges (degrees)
+      smooth_filter1->SetEdgeAngle(15);           // Not sure what this controls (degrees)
+      smooth_filter1->BoundarySmoothingOff();
+      smooth_filter1->NonManifoldSmoothingOff();
+      smooth_filter1->NormalizeCoordinatesOn();  // "Improves numerical stability"
+      smooth_filter1->Update();
+      mesh_filtered1 = smooth_filter1->GetOutput();
+
+      // Apply Laplacian Smoothing
+      // This moves the coordinates of each point toward the average of its adjoining points
+      vtkSmartPointer<vtkSmoothPolyDataFilter> smooth_filter2 = vtkSmartPointer<vtkSmoothPolyDataFilter>::New();
+      smooth_filter2->SetInputConnection(smooth_filter1->GetOutputPort());
+      //  smooth_filter2->SetInputData(mesh_cleaned);
```

Comment:
Last time I point this out, but all commented code in this PR should be removed unless you leave a comment describing why you want to leave it temporarily there

Reply:
This was just there for debugging, but these can come out.